sphinx/domains/std.py

```diff
@@ -782,11 +782,15 @@ def _resolve_doc_xref(self, env, fromdocname, builder, typ, target, node, contno
         if docname not in env.all_docs:
             return None
         else:
+            title = clean_astext(env.titles[docname])
             if node['refexplicit']:
                 # reference with explicit title
                 caption = node.astext()
+                if '{name}' in caption or '{number}' in caption:
+                    number = env.toc_secnumbers[docname].get('')
```

Comment:
I think it is better to notice if the document does not have section numbers.

Reply:
...do you mean to add a warning here if `number` is `None`? (Otherwise I don't think I understand your comment.)

Reply:
That's right